client/src/components/Allocations/AllocationsModals/AllocationsManageTeamTable/AllocationsTASProjectRoleSelector.jsx

```diff
@@ -0,0 +1,74 @@
+import React, { useState, useEffect } from 'react';
+import { useMutation } from 'react-query';
+import { useDispatch } from 'react-redux';
+import DropdownSelector from '_common/DropdownSelector';
+import { fetchUtil } from 'utils/fetchUtil';
+import { Button } from '_common';
+import styles from './AllocationsManageTeamTable.module.scss';
+
+const setProjectRole = async (projectId, userId, role) => {
+  const roleMap = {
+    Standard: 0,
+    Delegate: 2,
+  };
+  const url = `/api/users/tas-users/`;
+  const response = await fetchUtil({
+    url,
+    method: 'PUT',
+    body: JSON.stringify({
+      userId,
+      role: roleMap[role],
+      projectId,
+    }),
+  });
+  return response;
+};
+
+export const useProjectRole = (projectId, userId) => {
+  const dispatch = useDispatch();
+  const mutation = useMutation(async (role) => {
+    await setProjectRole(projectId, userId, role);
+    dispatch({
+      type: 'GET_PROJECT_USERS',
+      payload: { projectId },
+    });
+  });
+  return { mutation };
+};
+
+const TASProjectRoleSelector = ({ projectId, userId, role }) => {
+  const {
+    mutation: { mutate: setProjectRole, isLoading: isMutating, error },
+  } = useProjectRole(projectId, userId);
+  const [selectedRole, setSelectedRole] = useState(role);
+  useEffect(() => setSelectedRole(role), [role]);
+
+  if (error) return <span>Error</span>;
+
+  return (
+    <div style={{ display: 'inline-flex' }}>
+      <DropdownSelector
+        value={selectedRole}
+        onChange={(e) => setSelectedRole(e.target.value)}
+        disabled={isMutating}
+        className={styles['selector']}
+      >
+        <option value="Standard">Member</option>
+        <option value="Delegate">Allocation Manager</option>
+      </DropdownSelector>
+      {role !== selectedRole && (
+        <Button
+          type="secondary"
+          onClick={() => setProjectRole(selectedRole)}
+          size="small"
```

Comment:
@wesleyboar I noticed the loading spinner doesn't scale with the size of the button here, and is thus too big for this small button. Will address in 546 PR.

Reply:
Cool. Great.

<details>I saw in Sal's test video that the spinner exceeded button height. But when I tested, it fit. Although this button's spinner was larger compared ot button that was the spinner of the add button to the add button.</details>